### PR TITLE
[SYNC-114][FE] Various front-end editor style modifications

### PR DESCRIPTION
**Pull request by andyrochi**
## Description: 
- Update various styles to meet UX requirements
## Changes:
- Fix style of editor "Title" card
- Fix block card in editor
- Fix header style of side panel
- Add placeholder in text field
- Update custom.scss variables
- Fix font of text field
- Update edit page button variants
- Update citation design
- Add tiptap core & newer version of tiptap placehoder
- Update dependencies in yarn lock
- Fix dropdown button text styles
- Adjust editor page layout
- Update modal styles
## Test Scope:
-

### CODE REVIEW

File: src/assets/scss/post/main.scss [hunk 1]
```diff
@@ -23,62 +23,134 @@ li {
   }
 }
 
-%edit-no-outline {
-  color: $text-2;
-  background-color: $white;
-  padding: 0.25rem 1rem;
+%normal-btn {
```
Comment (jinan-tw) [inline src/assets/scss/post/main.scss:26]: jusy a question, what doesn % mean?

File: src/components/Editor/MenuBar.vue [hunk 2]
```diff
@@ -1,10 +1,12 @@
 <template>
   <div>
     <div class="menu-bar">
-      <template v-for="(item, index) in items">
-        <div v-if="item.type === 'divider'" :key="index" class="divider" />
-        <menu-item v-else :key="index" v-bind="item" />
-      </template>
+      <div class="left-section">
```
Comment (jinan-tw) [inline src/components/Editor/MenuBar.vue:4]: Kindly change the naming.

File: src/assets/scss/post/main.scss (hunk 1 above)
Comment (andyrochi) [reply src/assets/scss/post/main.scss:26]: It is a placeholder selector that the content can be included in other classes using the `@extend` keyword.
The content enclosed in % the selector will not appear in the output CSS if it is not `@extend`ed.

**Comment by jinan-tw** [comment]
image
tooltip position

File: src/components/Editor/MenuBar.vue (hunk 2 above)
Comment (andyrochi) [reply src/components/Editor/MenuBar.vue]: Done.

**Comment by jinan-tw** [comment]
AS IS
image
TO BE
have 123 in the text field

**Comment by jinan-tw** [comment]
AS IS
image
TO BE
Same to link feature, should have 123 in first text field

**Comment by jinan-tw** [comment]
AS IS
image
TO BE
The time should not be set if user click cancel, and add remove dateTime on it

**Comment by andyrochi** [comment]
The additional features that need to be fixed using javascript will be implemented in a further branch.

**Pull request merged**